docs/data/material/components/menus/NestedMenu.tsx

```diff
@@ -0,0 +1,287 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import MenuItem from '@mui/material/MenuItem';
+import {
+  Box,
+  ClickAwayListener,
+  Grow,
+  Icon,
+  MenuList,
+  Paper,
+  Popper,
+  Typography,
+} from '@mui/material';
+
+export default function BasicMenu() {
+  const MENU_LEVELS = 3;
```

Comment:
It would be great to have this in a form of a reusable component, so it's easy to copy and paste from our docs. So have a `<NestedMenu>` with all the logic that accepts the `options` prop.